subprojects/gradle/signer/src/main/kotlin/com/avito/plugin/SignArtifactTask.kt

```diff
@@ -8,22 +8,26 @@ import com.avito.logger.create
 import com.avito.time.DefaultTimeProvider
 import com.avito.time.TimeProvider
 import com.avito.utils.BuildFailer
+import okhttp3.HttpUrl
+import okhttp3.HttpUrl.Companion.toHttpUrl
 import org.gradle.api.DefaultTask
-import org.gradle.api.model.ObjectFactory
+import org.gradle.api.provider.Property
 import org.gradle.api.tasks.Input
 import org.gradle.api.tasks.TaskAction
-import org.gradle.kotlin.dsl.property
 import java.io.File
 import java.util.concurrent.TimeUnit
-import javax.inject.Inject
 
-abstract class SignArtifactTask @Inject constructor(objects: ObjectFactory) : DefaultTask() {
+@Suppress("UnstableApiUsage", "LeakingThis")
+abstract class SignArtifactTask : DefaultTask() {
 
-    @Input
-    val tokenProperty = objects.property<String>()
+    @get:Input
+    abstract val tokenProperty: Property<String>
 
-    @Input
-    val serviceUrl = objects.property<String>()
+    @get:Input
+    abstract val serviceUrl: Property<String>
+
+    @get:Input
+    abstract val readWriteTimeoutSec: Property<Long>
```

Comment:
It is an ad-hoc parameter but may be more reusable.
We could customize HTTP clients for each external service. And in all cases, a config may be the same.

The idea in properties:

```
avito.service.sign.http.readTimeoutSec=40
avito.service.sign.http.writeTimeoutSec=40

avito.service.another.http.readTimeoutSec=40
avito.service.another.http.writeTimeoutSec=40
```

Thus, I can override HTTP parameters for any service. Naming is a pattern that is supported in common HTTP models.

Why properties? Because we didn't agree yet to migrate to extensions in MBS-11039, it is only a hypothesis.

Reply:
i really don't like these implicit parameters, which is only known from documentation

Reply:
I understand the benefits of explicit config but don't know cases when we really need unique parameters for the same external service in different plugins.

Reply:
this `readWriteTimeoutSec` is bad example, 
MBS-11039 proposal with reusable nested extension blocks should be fine though!